### PR TITLE
Auto modules update

### PR DESCRIPTION
**Pull request by Sakib75**
## PR Type

PR_TYPE

## PR Checklist

- [ ] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [ ] A changelog entry has been made.
- [ ] Version number has been updated.
- [ ] Required modules have been added to respective "requirements*.txt" files.
- [ ] Relevant Test Cases added to this description (below).
- [ ] (Team) Label with affected action categories and semver status.

## Overview

Get the list of outdated modules and save the list into a json. In the next run, the saved outdated modules are updated to latest version

## Test Cases
- [TEST-0000](https://zeuz.zeuz.ai/Home/ManageTestCases/Edit/TEST-0000/)

### CODE REVIEW

File: Framework/module_installer.py [hunk 1]
```diff
@@ -82,6 +86,41 @@ def install_missing_modules():
...
+        
+        # Upgrading outdated modules found in last run
+        outdated_modules_filepath = os.path.dirname(os.path.abspath(__file__)) + os.sep + 'outdated_modules.json'
+        needs_to_be_updated = None
+        if(os.path.exists(outdated_modules_filepath)):
+            try:
+                needs_to_be_updated = json.load(open(outdated_modules_filepath))
+            except:
+                traceback.print_exc()
+            if(needs_to_be_updated):
```
Comment (sazid) [inline Framework/module_installer.py]: Its easier to read when you remove the braces around conditions, specially in Python.

**Review by sazid — CHANGES_REQUESTED** [review]
Shouldn't we be deleting the outdated modules list once we've done the update?

**Comment by Sakib75** [comment]
> Shouldn't we be deleting the outdated modules list once we've done the update?

In general, after we're done with the update it saves an empty list to the json.
Do you want me to set the logic to - delete the json file when we get empty list for outdated modules?

**Comment by sazid** [comment]
> > Shouldn't we be deleting the outdated modules list once we've done the update?
> 
> In general, after we're done with the update it saves an empty list to the json. Do you want me to set the logic to - delete the json file when we get empty list for outdated modules?

No that's okay if it sets it back to the empty list.
I'm merging this.

File: Framework/module_installer.py (hunk 1 above)
Comment (Sakib75) [reply Framework/module_installer.py]: Removed the extra braces

**Review by sazid — APPROVED** [review]

**Pull request merged**